internal/provider/apikey_resource.go

```diff
@@ -0,0 +1,466 @@
+					keyType := key.Type.MustGet()
+					if keyType == api.ApiKeyResponseType("publishable") {
+						hasDefaultPublishable = true
+					}
+				}
+			}
+		}
+	}
+
+	if !hasDefaultPublishable {
+		httpRespDefaultPublishable, errDefaultPublishable := client.V1CreateProjectApiKeyWithResponse(ctx, plan.ProjectRef.ValueString(), &api.V1CreateProjectApiKeyParams{Reveal: &reveal}, api.CreateApiKeyBody{
+			Name:              "default",
+			Type:              "publishable",
+			Description:       nullable.Nullable[string]{},
+			SecretJwtTemplate: nullable.Nullable[map[string]interface{}]{},
+		})
+		if errDefaultPublishable != nil {
+			msg := fmt.Sprintf("Unable to create default publishable apiKey, got error: %s", errDefaultPublishable)
+			return diag.Diagnostics{diag.NewErrorDiagnostic("Client Error", msg)}
+		}
+		if httpRespDefaultPublishable.JSON201 == nil {
+			msg := fmt.Sprintf("Unable to create default publishable apiKey, got status %d: %s", httpRespDefaultPublishable.StatusCode(), httpRespDefaultPublishable.Body)
+			return diag.Diagnostics{diag.NewErrorDiagnostic("Client Error", msg)}
+		}
+	}
+
+	// 2. Create apiKey
+	httpResp, err := client.V1CreateProjectApiKeyWithResponse(ctx, plan.ProjectRef.ValueString(), &api.V1CreateProjectApiKeyParams{Reveal: &reveal}, api.CreateApiKeyBody{
+		Name:              plan.Name.ValueString(),
+		Type:              "secret",
+		Description:       nullable.Nullable[string]{},
+		SecretJwtTemplate: nullable.NewNullableWithValue(map[string]interface{}{"role": "service_role"}),
+	})
+
+	if err != nil {
+		msg := fmt.Sprintf("Unable to create apiKey, got error: %s", err)
+		return diag.Diagnostics{diag.NewErrorDiagnostic("Client Error", msg)}
+	}
+	if httpResp.JSON201 == nil {
+		msg := fmt.Sprintf("Unable to create apiKey, got status %d: %s", httpResp.StatusCode(), httpResp.Body)
+		return diag.Diagnostics{diag.NewErrorDiagnostic("Client Error", msg)}
+	}
+
+	// Update computed fields from creation response
+	if httpResp.JSON201.Id.IsSpecified() && !httpResp.JSON201.Id.IsNull() {
+		plan.Id = types.StringValue(httpResp.JSON201.Id.MustGet())
+	} else {
+		plan.Id = types.StringNull()
+	}
+
+	if httpResp.JSON201.ApiKey.IsSpecified() && !httpResp.JSON201.ApiKey.IsNull() {
+		plan.ApiKey = types.StringValue(httpResp.JSON201.ApiKey.MustGet())
+	} else {
+		plan.ApiKey = types.StringNull()
+	}
+
+	if httpResp.JSON201.Type.IsSpecified() && !httpResp.JSON201.Type.IsNull() {
+		plan.Type = types.StringValue(string(httpResp.JSON201.Type.MustGet()))
+	} else {
+		plan.Type = types.StringNull()
+	}
+
+	obj, diags := types.ObjectValue(secretJwtTemplateAttrTypes, map[string]attr.Value{
+		"role": types.StringValue("service_role"),
+	})
+	if diags.HasError() {
+		return diags
+	}
+	plan.SecretJwtTemplate = obj
+
+	if diag := readApiKeyDatabase(ctx, plan, client); diag.HasError() {
+		for _, err := range diag.Errors() {
+			tflog.Warn(ctx, fmt.Sprintf("%s: %s", err.Summary(), err.Detail()))
+		}
+	}
+	return nil
```

Comment:
```suggestion
	return readApiKeyDatabase(ctx, plan, client)
```

If we were able to successfully create or update a key but couldn't read it back, I think it's safe to abort and let the user know